### PR TITLE
Implement correctly ordered destruction hooks.

### PR DESCRIPTION
**Pull request by rwjblue**
/cc @krisselden

### CODE REVIEW

**Comment by rwjblue** [comment]
Discussed with @krisselden and while there is still work to be done in Glimmer to remove the need to override some of these things this is still a step forward and we can iterate on it...

File: packages/ember-glimmer/lib/environment.js [hunk 1]
```diff
@@ -69,6 +69,9 @@ export default class Environment extends GlimmerEnvironment {
     super(...arguments);
     this.owner = owner;
 
+    // can be removed once https://github.com/tildeio/glimmer/pull/305 lands
+    this.destroyedComponents = undefined;
```
Comment (chancancode) [inline packages/ember-glimmer/lib/environment.js]: I thought you indented to just use this.destructors since you merged that PR?

Comment (chancancode) [inline packages/ember-glimmer/lib/environment.js]: ah we haven't merged yet. ignore me

Comment (rwjblue) [inline packages/ember-glimmer/lib/environment.js]: Right, I'd rather have a small kludge here in Ember for beta.1 and fix upstream in that other PR for beta.2.

**Pull request merged**